### PR TITLE
Distributed implementation

### PR DESCRIPTION
**Pull request by matthiasdiener**
Continued from https://github.com/kaushikcfd/pytato/pull/1

See #55.

### CODE REVIEW

File: pytato/transform.py [hunk 1]
```diff
@@ -164,6 +164,9 @@ def map_dict_of_named_arrays(self,
         return DictOfNamedArrays({key: self.rec(val.expr)
                                   for key, val in expr.items()})
 
+    def map_distributed_send(self, expr: DistributedSend) -> DictOfNamedArrays:
```
Comment (matthiasdiener) [inline pytato/transform.py:167]: Create deep copy of object

**Comment by matthiasdiener** [comment]
- Create new Mapper to partition graph

File: pytato/transform.py [hunk 2]
```diff
@@ -229,6 +232,9 @@ def map_dict_of_named_arrays(self, expr: DictOfNamedArrays) -> R:
         return self.combine(frozenset([expr]), *(self.rec(ary.expr)
                                                  for ary in expr.values()))
 
+    def map_distributed_send(self, expr: DistributedSend) -> DictOfNamedArrays:
```
Comment (matthiasdiener) [inline pytato/transform.py:235]: (see also CombineMapper in pymbolic) returns set of ?

**Comment by inducer** [comment]
Likely to be superseded by #65.

**Pull request closed without merging**